core/src/test/java/org/elasticsearch/index/engine/InternalEngineTests.java

```diff
@@ -280,6 +280,12 @@ public EngineConfig copy(EngineConfig config, EngineConfig.OpenMode openMode, An
     @After
     public void tearDown() throws Exception {
         super.tearDown();
+        if (engine != null && engine.isClosed.get() == false) {
```

Comment:
yeah, I was going back and forth between static maps in the assertion and how it's done now. The problem with static maps is that you need to clean them and it becomes ugly. I followed the same pattern we have in other places. The good news is that we also have file system leakage check and when people forget to close a snapshot it trips.